src/IterationTraits.php

```diff
@@ -225,16 +225,43 @@ public static function sortKeys(Iterator $iterator, $fn = null) {
      * Group all the the values into an array and return the result as a Sequence.
      *
      * @param Iterator $iterator
-     * @param callable  $fnToGroup
+     * @param callable $fnToGroup
+     * @param null $keys
      * @return Sequence
      */
-    public static function groupBy(Iterator $iterator, $fnToGroup) {
-        return self::wrapFunctionIntoSequenceOnDemand(function() use ($iterator, $fnToGroup) {
+    public static function groupBy(Iterator $iterator, $fnToGroup, $keys = null) {
```

Comment:
I am wondering if it makes more sense to have this just be $init or $fnInit.  Which is more powerful because it allows for richer options.